### ISSUE TITLE
ng deploy shows success message even if app building fails

### ISSUE THREAD

**Issue by EdricChan03** [angular-schule/angular-cli-ghpages#74]
Even if there's a problem in my code and the Angular Compiler detects the issue, the deploy builder assumes that all of the commands have run successfully:

```
📦 Building "rss-reader". Configuration: "production". Your base-href: "/rss-reader/"

chunk {0} runtime-es2015.b2aca5be9e7b8cc1a1b4.js (runtime) 1.41 kB [entry] [rendered]
chunk {1} main-es2015.e979d12557647e8d62fa.js (main) 128 bytes [initial] [rendered]
chunk {2} polyfills-es2015.270671f3654e232e0bbe.js (polyfills) 130 bytes [initial] [rendered]
chunk {3} styles.7d2e006da3faa06dd951.css (styles) 249 kB [initial] [rendered]

ERROR in app/components/release-notes/release-notes.component.ts:54:5 - error TS2322: Type 'Release' is not assignable to type 'string'.

54     return this.releaseNotes.releases[version];
chunk {scripts} scripts.df42bf43cef21311bf2d.js (scripts) 20.5 kB [entry] [rendered]
       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Date: 2019-09-03T14:02:29.664Z - Hash: ac8c61e38fe1b05d4fdf - Time: 15741ms
app/app.component.ts:122:7 - error TS2345: Argument of type '{ id: string; title: string; icon: string; showAsAction: false; onClickListener: () => any; }' is not assignable to parameter of type 'ActionItem'.
  Object literal may only specify known properties, and 'id' does not exist in type 'ActionItem'.

122       id: 'keyboard-shortcuts-action-item',
          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
app/app.component.ts:126:43 - error TS2551: Property 'openHotkeyHelpDialog' does not exist on type 'HotkeysService'. Did you mean 'hotkeyHelpDialogId'?

126       onClickListener: () => this.hotkeys.openHotkeyHelpDialog()
                                              ~~~~~~~~~~~~~~~~~~~~

  app/hotkeys/hotkeys.service.ts:36:12
    36   readonly hotkeyHelpDialogId = 'hotkey-help-dialog';
                  ~~~~~~~~~~~~~~~~~~
    'hotkeyHelpDialogId' is declared here.

index.html could not be copied to 404.html. This does not look like an angular-cli project?!
(Hint: are you sure that you have setup the directory correctly?)
🚀 Successfully published via angular-cli-ghpages! Have a nice day!
```

Ideally, the deploy builder should show that there's something wrong with the code above and stop the deploying process and/or not show the message that the site was successfully published.

**Comment by JohannesHoppe** [comment]
This is a good point. We should stop on error.

**Comment by EdricChan03** [comment]
This can be a problem in CI environments with a status (e.g. GitHub Actions) which may show a check even if there's an actual build issue in the change you pushed to the code, making it misleading.

**Comment by JohannesHoppe** [comment]
I absolutely agree. Until I, somebody else or you 😉 have figured out a fix, I suggest that every CI env does the following steps in advance:

```
ng lint
ng test
```

... this way `ng deploy` will be never called on broken code

(written on a mobile phone)

**Comment by nitoloz** [comment]
Have experienced this issue as well, but for it seems to be caused by wrong github credentials. So the flow looks following: build success => deploy starts => asks for credentials for 2 times => shows success message however nothing is deployed.

I would be amazing to see if smth goes wrong in deployment process.

**Comment by JohannesHoppe** [comment]
Okay, this should be an error. I will have to research on that...

**Comment by JohannesHoppe** [comment]
fixed via #85 by @masaxsuzu

**Comment by JohannesHoppe** [comment]
> So the flow looks following: build success => deploy starts => asks for credentials for 2 times => shows success message however nothing is deployed.
> 
> I would be amazing to see if smth goes wrong in deployment process.

This was fixed in https://github.com/angular-schule/angular-cli-ghpages/commit/6a8a3c217375b423a5654c388299bf3a12f9e375 (use callback to retrieve error message of gh-pages)
